test/game_connection/connector/test_dread_remote_connector.py

```diff
@@ -0,0 +1,81 @@
+import asyncio
+from unittest.mock import MagicMock, AsyncMock
+from PySide6 import QtCore
+
+import pytest
+from randovania.game_connection.connector.dread_remote_connector import DreadRemoteConnector
+from randovania.game_connection.executor.dread_executor import DreadExecutor, DreadExecutorToConnectorSignals, DreadLuaException, PacketType
+from randovania.games.game import RandovaniaGame
+
+
+@pytest.fixture(name="connector")
+def dread_remote_connector():
+    executor_mock = MagicMock(DreadExecutor)
+    executor_mock.signals = MagicMock(DreadExecutorToConnectorSignals)
+    connector = DreadRemoteConnector(executor_mock)
+    return connector
+
+async def test_general_class_content(connector: DreadRemoteConnector):
+    assert connector.game_enum == RandovaniaGame.METROID_DREAD
+    assert connector.game_enum == RandovaniaGame.METROID_DREAD
+    assert connector.description() == RandovaniaGame.METROID_DREAD.long_name
+
+    connector.Finished = MagicMock(QtCore.SignalInstance)
+    connector.connection_lost()
+    connector.Finished.emit.assert_called_once()
+
+    await connector.force_finish()
+    connector.executor.disconnect.assert_called_once()
+
+    connector.executor.is_connected = MagicMock()
+    connector.executor.is_connected.side_effect = [False, True]
+    assert connector.is_disconnected() == True
+    assert connector.is_disconnected() == False
+
+
+async def test_new_player_location(connector: DreadRemoteConnector):
+    connector.PlayerLocationChanged = MagicMock(QtCore.SignalInstance)
+
+    assert connector.inventory_index == -1
+    connector.inventory_index = 1
+    assert connector.inventory_index == 1
+
+    assert connector.current_region is None
+    connector.new_player_location_received("s010_cave")
+    connector.PlayerLocationChanged.emit.assert_called_once()
+    assert connector.current_region.name == "Artaria"
+
+    connector.PlayerLocationChanged = MagicMock(QtCore.SignalInstance)
+    connector.new_player_location_received("MAINMENU")
+    assert connector.inventory_index == -1
+    connector.PlayerLocationChanged.emit.assert_not_called()
```

Comment:
Do we want this to emit `None` as new player location if you switch from the game to main menu?

Reply:
That's kinda required for your connected state to change to "Title Screen" I believe.

Alternatively, just finish and let the GameConnector reconnect :P